ReText/window.py

```diff
@@ -943,8 +944,39 @@ def standardPrinter(self, title):
 		printer = QPrinter(QPrinter.HighResolution)
 		printer.setDocName(title)
 		printer.setCreator('ReText %s' % app_version)
+		page_size = self.getPageSizeByName(globalSettings.paperSize)
+		if page_size is not None:
+		    printer.setPaperSize(page_size)
+		else:
+		    QMessageBox.warning(self, '',
+				'Unrecognized PaperSize setting "{0}".'.format(
+				globalSettings.paperSize))
 		return printer
 
+	def getPageSizeByName(self, page_size_name):
+		""" Returns a validated PageSize instance
+		corresponding to the given name. Returns
+		None if the name is not a valid PageSize.
+		"""
+
+		page_size = None
+		match = None
+		if page_size_name:
+			match = list(filter(lambda y:y.lower()==page_size_name.lower(),
+						self.availablePageSizes()))
+		if match:
+			page_size = getattr(QPagedPaintDevice, match[0])
+		return page_size
+
+	def availablePageSizes(self):
+		""" List available page sizes.
+		"""
```

Comment:
Maybe use [QPrinter::availablePageSizes()](http://doc.qt.io/qt-5/qprinterinfo.html#supportedPageSizes) instead of this function? You will need to pass printer info to `getPageSizeByName`, but that should not be a problem.

Reply:
Hmm.  supportedPageSizes (possibly same as availablePageSizes ?) consistently returns empty array in my set-up.  Perhaps this feature is not quite working on my system or there are some version differences. I'm using python-qt5-5.7-7.fc25.x86_64 on Fedora  Linux 4.13.5-100.fc25.x86_64. Leaving as is for now as this seems compatible/working. Feel free to modify/